### PR TITLE
test: Adds tests to dnd controls

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Adds tests to dnd controls. 

@kgabryje We are auditing the codebase of Explore searching for components that don’t have sufficient test coverage (at least 80%). We are using Codecov reports to find the components. From what I understood this feature is still in POC mode, right? Anyway, I added some test files here to help with the test coverage and serve as a base for more complete tests. Can you please review it?  

PS: `DndMetricSelect.test.tsx` has only one test. After we add typescript support to  `DndMetricSelect` `props` we should complement this.

### TEST PLAN
1 - Execute all drag and drop tests.
2 - All tests should pass.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by junlincc** [comment]
@michael-s-molina Thanks for adding test to dnd, this feature is completed for phase 1 poc and is still evolving rapidly. there's no harm adding test though. 🙏  @kgabryje please review with a grey check mark, hopefully it turns green in days. 😉

File: superset-frontend/src/explore/components/controls/DndColumnSelectControl/OptionWrapper.test.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import { render, screen, fireEvent } from 'spec/helpers/testing-library';
+import { DndItemType } from 'src/explore/components/DndItemType';
+import OptionWrapper from 'src/explore/components/controls/DndColumnSelectControl/OptionWrapper';
+
+test('renders with default props', () => {
+  const { container } = render(
+    <OptionWrapper
+      index={1}
+      clickClose={jest.fn()}
+      type={'Column' as DndItemType}
```
Comment (kgabryje) [inline superset-frontend/src/explore/components/controls/DndColumnSelectControl/OptionWrapper.test.tsx:29]:
```suggestion
      type={DndItemType.Column}
```

Comment (kgabryje) [reply superset-frontend/src/explore/components/controls/DndColumnSelectControl/OptionWrapper.test.tsx:29]: Same for lines 47 and 55

**Review by pkdotson — APPROVED** [review]
LGTM!

File: superset-frontend/src/explore/components/controls/DndColumnSelectControl/OptionWrapper.tsx [hunk 2]
```diff
@@ -99,13 +105,8 @@ export default function OptionWrapper(
   drag(drop(ref));
 
   return (
-    <DragContainer ref={ref} {...props}>
-      <Option
-        index={index}
-        clickClose={clickClose}
-        onShiftOptions={onShiftOptions}
```
[the remark this answers is not in the document]
Comment (michael-s-molina) [reply superset-frontend/src/explore/components/controls/DndColumnSelectControl/OptionWrapper.tsx:106]: I checked when I removed it 😉. Nice observation though!

```
export const DragContainer = styled.div`
  margin-bottom: ${({ theme }) => theme.gridUnit}px;
  :last-child {
    margin-bottom: 0;
  }
`;
```

**Review by rusackas — APPROVED** [review]
LGTM!

**Pull request merged**